File: Golang/HyechatDemo/domain/eventhandler/eventHandler.go
package eventhandler

import (
	"context"
	"encoding/json"
	"net/http"
	"strconv"

	"HyechatDemo/heybotclient/common"
	"HyechatDemo/heybotclient/model"
)

type Domain struct {
	httpClient  *http.Client
	clientToken string
}

const (
	RepeaterCommandID = "9527"
)

func New(token string) *Domain {
	domain := &Domain{
		httpClient:  &http.Client{},
		clientToken: token,
	}
	return domain
}

func (d *Domain) OnMessage(c context.Context, msg *model.GenericType) error {
	switch msg.Type {
	case model.MsgTypeCommand:
		return d.OnUseBotCommand(c, msg.Data)
	default:
	}
	return nil
}

func (d *Domain) OnUseBotCommand(_ context.Context, data json.RawMessage) error {
	command := &model.UseCommandData{}
	if err := json.Unmarshal(data, command); err != nil {
		return err
	}
	if command != nil && command.CommandInfo != nil {
		commandID := command.CommandInfo.Id
		if commandID == RepeaterCommandID {
			if len(command.CommandInfo.Options) == 1 {
				option := command.CommandInfo.Options[0]
				if option.Type == model.TypeString {
					return common.SendChannelIM(d.httpClient, &model.ChannelImSendReq{
						AtUserId:     strconv.Itoa(command.SenderInfo.UserId),
						Msg:          option.Value,
						ChannelId:    command.ChannelBaseInfo.ChannelId,
						MsgType:      10,
						ChannelType:  1,
						RoomId:       command.RoomBaseInfo.RoomId,
						Addition:     "{\"img_files_info\":[]}",
						HeychatAckId: "",
					}, d.clientToken)
				}
			}
		}
	}
	return nil
}

File: Python/HeyhcatDemo/eventhandler/eventhandler.py
import common.common
import conf.command as command
import conf.model as model

RepeaterCommandID = "1830851200470908928"


def on_use_bot_command(data):
    meta = {}
    if isinstance(data, dict):
        meta = model.UseCommandData(**data)
    if meta and meta.command_info:
        command_id = meta.command_info.id
        if command_id == RepeaterCommandID:
            on_repeater(meta)


def on_repeater(meta):
    if len(meta.command_info.options) == 1:
        option = meta.command_info.options[0]
        if option.type == command.TYPE_STRING:
            req = model.ChannelImSendReq(
                msg=option.value,
                msg_type=model.MSG_TYPE_MDTEXT,
                channel_id=meta.channel_base_info.channel_id,
                room_id=meta.room_base_info.room_id,
            )
            common.common.SendMessage(req)


class EventHandler:
    async def on_message(self, data):
        message_type = data["type"]
        message_data = data["data"]
        if message_type == model.MSG_TYPE_USECOMMAND:
            on_use_bot_command(message_data)
